common/web/keyboard-processor/src/keyboards/keyboard.ts

```diff
@@ -502,7 +502,7 @@ export default class Keyboard {
     // Set the flags for the state keys - for desktop devices. For touch
     // devices, the only state key in use currently is Caps Lock, which is set
     // when the 'caps' layer is active in ActiveKey::constructBaseKeyEvent.
-    if(!Lkc.device.touchable) {
+    if(!Lkc.device?.touchable) {
```

Comment:
I think one of the input assumptions here is that `device` is valid. So this may be masking a more significant bug.

Reply:
With regard to the stack trace from the issue:

    TypeError: Cannot read properties of null (reading 'baseKeyEvent')
    at i.constructKeyEvent (keyboard.ts:473:21)
    at i.constructBaseKeyEvent (activeLayout.ts:287:62)
    at i.polyfill (activeLayout.ts:239:10)

https://github.com/keymanapp/keyman/blob/4c7b7992da6491dc5bf30245dc5851f8c1d471c1/common/web/keyboard-processor/src/keyboards/activeLayout.ts#L283-L293

It appears this code path went unused and untested until the bug was observed.  Now, I think I did do a lot of key-event preprocessing code refactoring in 17.0 during ES module work, so this is probably 17.0-only.  All in all, the bug is probably a "loose end" I accidentally didn't properly handle during that initiative.